qcodes/instrument_drivers/oxford/kelvinox.py

```diff
@@ -0,0 +1,534 @@
+
+    def _set_valve_status(self, status, valve):
+        """
+        Return the status of the valve number "valve" where valve must be a number between 1 and 20 (self._map_valve)
+        status: 0 for off and 1 for on
+        """
+        self.remote()
+        log.info('Setting valve %s status' % self._valve_map[valve])
+        self._execute('P%i' % (2 * valve + numpy.mod(status + 1, 2)))
+        self.local()
+
+    def _set_V6_valve(self, status):
+        """
+        This set the opening of the stepper valve 6. Status should be a percentage.
+        """
+        self.remote()
+        log.info('Setting valve 6 status')
+        self._execute('G%i' % int(10 * status))
+        self.local()
+
+    def _get_V6_valve(self):
+        """
+        Return the opening of valve 6.
+
+        Output:
+            result(float): Opening of V6 valve in percent
+        """
+        result = self._execute('R7')
+        return float(result.replace('R', '')) / 10
+
+    def _set_V12A_valve(self, status):
+        """
+        This set the opening of the stepper valve 12. Status should be a percentage.
+        """
+        self.remote()
+        log.info('Setting valve V12A status')
+        self._execute('H%i' % int(10 * status))
+        self.local()
+
+    def _get_V12A_valve(self):
+        """
+        Return the opening of valve V12A.
+
+        Output:
+            result(float): Opening of valve V12A in percent
+        """
+        result = self._execute('R8')
+        return float(result.replace('R', '')) / 10
+
+    def rotate_Nvalve(self, value):
+        """
+        This set the opening of the stepper valve N. Status should be a percentage.
+        """
+        self.remote()
+        log.info('Setting valve N status')
+        value = int(value)
+        if value < 1000:
+            if value > -1:
+                self._execute('N%i' % int(value))
+                self.local()
+            else:
+                print('Wrong value....')
+        else:
+            print('Wrong value....')
+
+    def _get_still_sorb_status(self):
+        """Get the the still and sorb status
+            O0 : both off
+            O1 : Still ON, sorb OFF
+            O2 : Still OFF, sorb on T control
+            O3 : Still ON, sorb ON T control
+            O4 : Still OFF, sorb on power control
+            O5 : Still ON, sorb ON power control
+        """
+        result = self._execute('X')
+        result = result[17:19]
+        return result
+
+    def _set_still_sorb_status(self, status):
+        """Get the the still and sorb status
+            O0 : both off
+            O1 : Still ON, sorb OFF
+            O2 : Still OFF, sorb on T control
+            O3 : Still ON, sorb ON T control
+            O4 : Still OFF, sorb on power control
+            O3 : Still ON, sorb ON power control
+        """
+        self.remote()
+        log.info('Set still and sorb status')
+        self._execute(status)
+        self.local()
```

Comment:
This is the way it was operated before and we don't have a reason to change it, so I think it is best to leave it like it is.